Changed big flowers so they can only be placed on dirt

Altenius opened the pull request:
Fixes #3559

Altenius commented:
Looks like Travis CI failed because apt-get install couldn't find a package.

madmaxoft commented on src/Blocks/BlockBigFlower.h:
```diff
@@ -77,7 +77,10 @@ class cBlockBigFlowerHandler :
-		return ((a_RelY > 0) && (a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ) != E_BLOCK_AIR) && (a_RelY < cChunkDef::Height - 1) && ((a_Chunk.GetBlock(a_RelX, a_RelY + 1, a_RelZ) == E_BLOCK_AIR) || (a_Chunk.GetBlock(a_RelX, a_RelY + 1, a_RelZ) == E_BLOCK_BIG_FLOWER)));
+		BLOCKTYPE BlockType;
+		NIBBLETYPE BlockMeta;
+		a_Chunk.GetBlockTypeMeta(a_RelX, a_RelY - 1, a_RelZ, BlockType, BlockMeta);
```
This will fail if `a_RelY` < 1

Overall, the condition below is really complicated, refactor it using standalone `if` statements to make it more readable.

Altenius replied:
What do you mean by standalone if statements? Why not break it into multiple lines?

Altenius commented:
I fixed the `a_RelY == 0` code and fixed another problem where tall flowers can only be placed if there's air above them, which isn't the case in vanilla.

madmaxoft commented on src/Blocks/BlockBigFlower.h:
```diff
@@ -77,7 +77,15 @@ class cBlockBigFlowerHandler :
+		NIBBLETYPE BlockMeta;
+		a_Chunk.GetBlockTypeMeta(a_RelX, a_RelY - 1, a_RelZ, BlockType, BlockMeta);
+
+		return IsBlockTypeOfDirt(BlockType) || ((BlockType == E_BLOCK_BIG_FLOWER) && !(BlockMeta & 0x08));
```
Style: Replace `!(BlockMeta & 0x08)` with an explicit comparison - we only allow bool values in conditions, the result of that expression is NOT a bool value.

Perhaps even better, replace it with a call to new function `IsMetaTopPart` - adds proper meaning and is much more maintainable.

madmaxoft commented:
Wait, does this mean that a tall flower can drill through any block above it? So it would be possible to break above the Nether in vanilla?

QuImUfu commented:
No, in vanilla you can't place duble plants, if there is something else than air in the upper block.

Altenius commented:
What I meant was flowers can be placed if there's air above the top part.
